src/main/java/org/broadinstitute/hellbender/tools/AnnotatePairOrientation.java

```diff
@@ -0,0 +1,119 @@
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    protected File outputFile;
+
+    public final static String CUTOFF_SHORT_NAME = "c";
+    public final static String CUTOFF_LONG_NAME = "meanBaseQualityCutoff";
+    public final static double MEAN_BASE_QUALITY_DEFAULT_CUTOFF = 7.0;
+    @Argument(
+            doc = "Cutoff for the mean base quality value(s) to count the read.  These are for bases that overlap the variant.",
+            shortName = CUTOFF_SHORT_NAME, fullName = CUTOFF_LONG_NAME, minValue = 0.0, maxRecommendedValue = 10.0,
+            optional = true
+    )
+    private Double meanBaseQualityCutoff = MEAN_BASE_QUALITY_DEFAULT_CUTOFF;
+
+    private VariantContextWriter vcfWriter;
+
+    @Override
+    public void onTraversalStart() {
+        vcfWriter = createVCFWriter(outputFile);
+        vcfWriter.writeHeader(createVCFHeader(getHeaderForVariants(), getCommandLine()));
+    }
+
+    @Override
+    public List<ReadFilter> getDefaultReadFilters() {
+        return Mutect2Engine.makeStandardMutect2ReadFilters();
+    }
+
+    @Override
+    public void apply(VariantContext variant, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+
+
+        final ReadPileup readPileup = GATKProtectedVariantContextUtils.getPileup(variant, readsContext);
+        final List<Genotype> updatedGenotypes = new ArrayList<>();
+
+        final Map<String, ReadPileup> sampleToReadPileup = readPileup.splitBySample(getHeaderForReads(), null);
+
+        for (Genotype g : variant.getGenotypes()) {
+
+            final ReadPileup genotypeSamplePileup = sampleToReadPileup.get(g.getSampleName());
+            final GenotypeBuilder gb = new GenotypeBuilder(g);
+            OxoGReadCounts.annotateSingleVariant(variant, gb, genotypeSamplePileup, meanBaseQualityCutoff);
+            updatedGenotypes.add(gb.make());
+        }
+
+        final VariantContextBuilder vcb = new VariantContextBuilder(variant);
+        vcb.genotypes(updatedGenotypes);
+        vcfWriter.add(vcb.make());
+    }
+
+    @Override
+    public boolean requiresReads() {
+        return true;
+    }
+
+    private static VCFHeader createVCFHeader(final VCFHeader inputVCFHeader, final String commandLine) {
+        Utils.nonNull(inputVCFHeader);
+
+        // Setup header for output file
+        final Set<VCFHeaderLine> headerLines = new LinkedHashSet<>(inputVCFHeader.getMetaDataInInputOrder());
+        headerLines.add(GATKVCFHeaderLines.getFormatLine(GATKVCFConstants.F1R2_KEY));
+        headerLines.add(GATKVCFHeaderLines.getFormatLine(GATKVCFConstants.F2R1_KEY));
+        headerLines.add(new VCFHeaderLine("command", commandLine));
+        final SampleList samples = new IndexedSampleList(inputVCFHeader.getGenotypeSamples());
+        final Set<String> sampleNameSet = samples.asSetOfSamples();
```

Comment:
I think you can inline this into the return statement.

Reply:
Done